Sources/Sentry/SentryTracer.m

```diff
@@ -92,6 +103,14 @@ - (instancetype)initWithTransactionContext:(SentryTransactionContext *)transacti
             initSlowFrames = currentFrames.slow;
             initFrozenFrames = currentFrames.frozen;
         }
+#endif
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        [profilerLock lock];
+        if (profiler == nil && [_hub getClient].options.enableProfiling) {
```

Comment:
`l`: We could check if `enableProfiling` is enabled and avoid unnecessary locks here and also below.

Reply:
Good point!